### ISSUE TITLE
--variant-input --only-use-input-alleles does not detect variants

### ISSUE THREAD

**Issue by EReckase** [ekg/freebayes#160]
Sample BAM:
https://www.dropbox.com/s/jafnsn33j0kzjf9/MW410_R1_combined.molbar.trimmed.deduped.variants.bam?dl=0

Sample VCF for variant-input:
```
##fileformat=VCFv4.1
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO
chr7    55242487        .       C       T       .       .       .
chr7    140453136       .       A       T       .       .       .
chr9    139391636       .       G       A       .       .       .
chr9    139397707       .       G       A       .       .       .
chr10   89711881        .       A       G       .       .       .
chr17   7578495 .       G       A       .       .       .
chr17   7578496 .       T       C       .       .       .
chr19   17495782        .       G       A       .       .       .
```

Variants are detected when the variant-input option is not specified.  Using additional options:

--ploidy 4 --min-base-quality 20 --min-alternate-fraction .02 --min-alternate-count 10 --no-indels --only-use-input-alleles --min-mapping-quality 20

**Comment by ekg** [comment]
You need to also specify --use-only-variant-input.

**Comment by EReckase** [comment]
I have been using version 0.9.16, is that parameter in that version?

**Comment by ekg** [comment]
So sorry, I misread the title.

I'll have to look into this.

**Comment by EReckase** [comment]
Additional information.  I tried the full BAM file and the variant input from above with the latest git code, and it wrote out the following:

```
getting variants in chr1:43814779-43815909
getting variants in chr1:43815909-115251846
getting variants in chr1:115251846-115252854
getting variants in chr1:115252854-115257020
getting variants in chr1:115257020-115258022
getting variants in chr1:115258022-115259226
getting variants in chr1:115259226-115260227
getting variants in chr1:115260227-162740791
getting variants in chr1:162740791-162742510
getting variants in chr1:162742510-162744374
getting variants in chr1:162744374-162745879
getting variants in chr1:162745879-162746882
getting variants in chr1:162746882-162749105
getting variants in chr1:162749105-162750408
getting variants in chr1:162750408-226253126
```

No variants were detected, and the only chromosome displayed on the output is chr1.  If I filter out only the chr7 reads from the bam and use the same command line options, I get some mutations:

```
getting variants in chr7:55210529-55211658
getting variants in chr7:55211658-55222223
getting variants in chr7:55222223-55223227
getting variants in chr7:55223227-55233290
getting variants in chr7:55233290-55234291
getting variants in chr7:55234291-55242179
getting variants in chr7:55242179-55243180
chr7	55242487	.	C	T	101505	.	AB=0;ABP=0;AC=4;AF=1;AN=4;AO=3340;CIGAR=1X;DP=3349;DPB=3349;DPRA=0;EPP=3394.57;EPPR=7.35324;GTI=0;LEN=1;MEANALT=2;MQM=41.9683;MQMR=42;NS=1;NUMALT=1;ODDS=1842.72;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=128128;QR=308;RO=8;RPL=2834;RPP=3526.51;RPPR=20.3821;RPR=506;RUN=1;SAF=2316;SAP=1088.27;SAR=1024;SRF=6;SRP=7.35324;SRR=2;TYPE=snp	GT:DP:RO:QR:AO:QA	1/1/1/1:3349:8:308:3340:128128
getting variants in chr7:55243180-55244265
getting variants in chr7:55244265-55249620
getting variants in chr7:55249620-55250633
getting variants in chr7:55250633-55259807
getting variants in chr7:55259807-55260810
getting variants in chr7:55260810-116340247
getting variants in chr7:116340247-116341248
getting variants in chr7:116341248-116403673
getting variants in chr7:116403673-116404678
getting variants in chr7:116404678-116412394
getting variants in chr7:116412394-116413395
getting variants in chr7:116413395-116418345
getting variants in chr7:116418345-116423840
getting variants in chr7:116423840-116424870
getting variants in chr7:116424870-116436524
getting variants in chr7:116436524-116437556
getting variants in chr7:116437556-128845662
getting variants in chr7:128845662-128846705
getting variants in chr7:128846705-128847706
getting variants in chr7:128847706-128850849
getting variants in chr7:128850849-128851856
getting variants in chr7:128851856-128852862
getting variants in chr7:128852862-140454166
chr7	140453136	.	A	T	11158.3	.	AB=0;ABP=0;AC=4;AF=1;AN=4;AO=361;CIGAR=1X;DP=361;DPB=361;DPRA=0;EPP=8.79087;EPPR=0;GTI=0;LEN=1;MEANALT=1;MQM=39.4017;MQMR=0;NS=1;NUMALT=1;ODDS=212.024;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=13883;QR=0;RO=0;RPL=122;RPP=85.3518;RPPR=0;RPR=239;RUN=1;SAF=242;SAP=94.0136;SAR=119;SRF=0;SRP=0;SRR=0;TYPE=snp	GT:DP:RO:QR:AO:QA	1/1/1/1:361:0:0:361:13883
getting variants in chr7:140454166-140481837
getting variants in chr7:140481837-140482850
getting variants in chr7:140482850-148509512
```

**Comment by EReckase** [comment]
Also, is there a way to suppress the 'getting variants' messages?

**Comment by EReckase** [comment]
I attempted to only include alignments in the bam that aligned to chromosomes that were present in the variant input file, but that also did not work.

**Comment by ekg** [comment]
OK, the getting variants messages are a bug... debugging message that
should have been removed.

I'll work from your test.

**Comment by ekg** [comment]
I'm having some trouble reproducing this.

Would you please upload the bgzipped VCF file you're using as input?

When I attempt to index the file, I get:

```text
[get_intv] the following line cannot be parsed and skipped: 7    55242487        .       C       T       .       .       .
[ti_index_core] the indexes overlap or are out of bounds
```
I attempted to resolve this by using `unexpand`, but I'm not sure the issue is related to tabs/spaces... weird. Any ideas?

**Comment by EReckase** [comment]
https://www.dropbox.com/s/d5mn6npmst3501t/SK-MEL-28.vcf.gz?dl=0

Thanks for looking.

**Comment by EReckase** [comment]
I've got a pretty tight deadline to get this working - if it's going to be more than a few days, can you let me know so I can code up a workaround?

**Comment by thondeboer** [comment]
Hi there...I am going to put in my request to get this fixed as well....Any chance this gets addressed any time soon? We are relying on this feature for an important part of our software but it's just not working...Thanks!

**Comment by ekg** [comment]
@EReckase @thondeboer OK, I'm on it.

@EReckase sorry if you've had to work around it. An easy solution is to run a job for each variant by transforming the list of variants in the VCF into a series of targets.

**Comment by EReckase** [comment]
This is not resolved; the only way to get --only-use-input-alleles to work is to only process a single chromosome of the input bam at a time.

**Comment by ekg** [comment]
If this is a blocking issue, why not just run without
--use-only-input-alleles and filter for only variants of interest
afterwards?

I wish there were funding to support a developer to help out. I am
stretched very thin as the sole maintainer and this is unfortunately
resulting in poor support for you. (I am also working on another project
full time and additionally carrying a PhD rotation project forward.)

Unlike other commercial systems (e.g. GATK) freebayes is 100% free for any
use. But this freedom comes at a cost as it must be up to the community to
contribute support to the system.

In other words I could really use a hand here.

That said, I will do what I can in the coming days. Apologies for the
trouble.

**Comment by EReckase** [comment]
Actually, we tried that.  Without specifying only use input alleles, we do not necessarily get a specific call/no call on each variant, which we needed for our report.  Additionally, sometimes a variant would be called within a specified variant (a SNP in the middle of a targeted deletion, for example) and then the specified variant would be missed.
To work around these, we had to do the per-chromosome processing, there wasn't any other way.  I do think that we can assist in development, I'll contact you shortly.

**Comment by ekg** [comment]
@EReckase 

> Actually, we tried that.

It's unfortunate this didn't work but the failure mode (that whole-chrom solves) does clarify what might have gone wrong.

> I do think that we can assist in development

This would be immensely appreciated!
